helm/charts/hydra/templates/secrets.yaml

```diff
@@ -5,8 +5,13 @@ metadata:
   namespace: {{ .Release.Namespace }}
   labels:
 {{ include "hydra.labels" . | indent 4 }}
+  annotations:
+    # Create the secret before installation, and only then. This saves the secret from regenerating during an upgrade
+    "helm.sh/hook": "pre-install"
+    "helm.sh/hook-delete-policy": "before-hook-creation"
 type: Opaque
 data:
-  secretsSystem: {{ include "hydra.secrets.system" . | required "Value secrets.system can not be empty!" | b64enc | quote }}
-  secretsCookie: {{ include "hydra.secrets.cookie" . | required "Value secrets.cookie can not be empty!" | b64enc | quote }}
+  # Generate a random secret if the user doesn't give one. User given password has priority
+  secretsSystem: {{ ( include "hydra.secrets.system" . | default ( randAlphaNum 32 )) | required "Value secrets.system can not be empty!" | b64enc | quote }}
```

Comment:
Is this cryptographically secure?

Reply:
what do you mean by that?

Reply:
Are values coming from `randAlphaNum` cryptographically secure? Meaning that they come from e.g. `/dev/urandom` as opposed to a PRNG

Reply:
Took me a while but found the [docs](https://github.com/Masterminds/sprig/blob/40a817194fa6bb2cd370b302c30b27d3706f122d/docs/strings.md#randalphanum-randalpha-randnumeric-and-randascii) for this function, seems we are safe :)

Reply:
Awesome!